pkgs/applications/misc/pagefind/default.nix

```diff
@@ -32,7 +33,15 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-pcgcu9zylSTjj5rxNff+afFBWVpN5sGtlpadG1wb93M=";
   };
 
-  cargoHash = "sha256-E4gjG5GrVWkMKgjQiAvEiSy2/tx/yHKe+5isveMZ9tU=";
+  cargoPatches = [
+    (fetchpatch { # https://github.com/CloudCannon/pagefind/pull/680
+      name = "cargo-update-time.patch";
+      url = "https://github.com/CloudCannon/pagefind/commit/e6778572d225316803180db822f5cc12a936acd2.patch";
```

Comment:
Please don't use fetchpatch for non-upstream commits, because they might not stick around.  It's better to include the patches in Nixpkgs in this case.

Reply:
This is news to me, i've yet to experience github to stop resolving the commit even if the PR is closed and the branch deleted, and a plethora of packages in nixpkgs fetch commits from open PRs. 

There is also the recommendation from https://github.com/NixOS/nixpkgs/tree/c55e2908deb848d0e94d8170cb4b0a1e9bc30c8c/pkgs#reviewing-contributions which reads:

> patches that are remotely available are fetched rather than vendored

Reply:
GitHub will preserve it if the PR is closed and branch deleted.  What they won't do is preserve it if the PR is force pushed.  You can see this by going to any old Nixpkgs PR that was force pushed (arbitrary example: PR 123456), and trying to view the commits that preceded the final force push.

Reply:
It got merged, so we can continue fetching it.

Reply:
Thanks for your explanation alyssais, it does seemt be the case. I was convinced of the opposite since I never ran into such a case while authoring https://github.com/NixOS/nixpkgs/pull/258431, even though I dug through the timeline of multiple old PRs to locate the correct commit refs.

Reply:
If I'm successful with the documentation update I currently have in flight, maybe next I'll see if I can get the documentation updated to explain this once and for all. :)